### DISCUSSION TITLE
Multiple cameras per printer support?

### DISCUSSION THREAD

**Discussion by KalixtShawxo** [OctoFarm/OctoFarm#772] [Feature Request]
So on my printers, I have 2 cameras each. a raspi cam and a usb webcam.  they look at the print bed from the side and above (respectively on each).  is there a way to have the Octofarm recognize both cameras? and maybe have them both available for view on the camera page instead of picking just one or the other? 

If not currently, is it something that could be added easily?

thanks.

**Comment by NotExpectedYet** [comment]
> So on my printers, I have 2 cameras each. a raspi cam and a usb webcam. they look at the print bed from the side and above (respectively on each). is there a way to have the Octofarm recognize both cameras? and maybe have them both available for view on the camera page instead of picking just one or the other?
> 
> If not currently, is it something that could be added easily?
> 
> thanks.

The main issue for me is where to put them in the UI... 

I had a few ideas, the camera urls can go into an Array and then 

1. there's a carousell on the camera field to allow switching between both at will. 

2. The camera field showed half of each camera allowing to click into both full view. 

3. Create a second printer box next to the first one with the second camera on. 

None of these ideas feel great to me, do you have any ideas or suggestions on how you would see this feature?

**Comment by KalixtShawxo** [comment]
Hi James, What I was thinking about was in the printer setup, where it has the address URL for the printer,  add a second box for an optional second camera address, and then next to each is a checkbox for "display feed?".This way, you could  setup both cameras for the printer, and have either or both displayed on the cameras page. I would expect side by side, (normal size) mixed amongst the other cameras for whatever other printers you have configured. 
Maybe have a button on the camera page off to the side with an override "show all cameras" option? 
That's trying to look at it from a perspective of "bigger than what I have".I have 2 printers, 2 cameras each, I would display all 4. 

Sent from Yahoo Mail on Android 
 
  On Mon, Nov 22, 2021 at 2:24, James ***@***.***> wrote:   



So on my printers, I have 2 cameras each. a raspi cam and a usb webcam. they look at the print bed from the side and above (respectively on each). is there a way to have the Octofarm recognize both cameras? and maybe have them both available for view on the camera page instead of picking just one or the other?

If not currently, is it something that could be added easily?

thanks.


The main issue for me is where to put them in the UI...

I had a few ideas, the camera urls can go into an Array and then
   
   -    
there's a carousell on the camera field to allow switching between both at will.

   -    
The camera field showed half of each camera allowing to click into both full view.

   -    
Create a second printer box next to the first one with the second camera on.


None of these ideas feel great to me, do you have any ideas or suggestions on how you would see this feature?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by NotExpectedYet** [comment]
> Hi James, What I was thinking about was in the printer setup, where it has the address URL for the printer,  add a second box for an optional second camera address, and then next to each is a checkbox for "display feed?".This way, you could  setup both cameras for the printer, and have either or both displayed on the cameras page. I would expect side by side, (normal size) mixed amongst the other cameras for whatever other printers you have configured.  Maybe have a button on the camera page off to the side with an override "show all cameras" option?  That's trying to look at it from a perspective of "bigger than what I have".I have 2 printers, 2 cameras each, I would display all 4. 

I hadn't considered a setting there, that's a good point! My original thinking was there's going to be someone who says, I want 3! xD then where does that stop. I do think limiting it to a secondary camera would cover most basis if not all. 

It makes sense to have a switch, toggle between and all cameras. 

There's some other camera work I'm planning to sort out, ie... the aweful fullscreen implementation that's available now. I will add this to the list to consider when I get around to that. It's probably going to be after V1.2 which I'm working on at the moment. I'd add it but it's already a rather large overhaul and I wanna get it out the door ahaha.

**Comment by KalixtShawxo** [comment]
Can you even do three cameras? I guess you could do 2 USB and a RasPi. I never thought about it. But yeah, a limit of two would be good since most people probably self limit to 2 cause the resources needed for more pretty much limits you to a high end Pi4. I have two 3B+ and they handle it ok. 
Thanks for considering the feature, and no rush. 😁Knowing it's a future possibility is good. 
When you get around to it, I'll happily beta for you. 

Sent from Yahoo Mail on Android 
 
  On Mon, Nov 22, 2021 at 6:35, James ***@***.***> wrote:   



Hi James, What I was thinking about was in the printer setup, where it has the address URL for the printer, add a second box for an optional second camera address, and then next to each is a checkbox for "display feed?".This way, you could setup both cameras for the printer, and have either or both displayed on the cameras page. I would expect side by side, (normal size) mixed amongst the other cameras for whatever other printers you have configured. Maybe have a button on the camera page off to the side with an override "show all cameras" option? That's trying to look at it from a perspective of "bigger than what I have".I have 2 printers, 2 cameras each, I would display all 4.


I hadn't considered a setting there, that's a good point! My original thinking was there's going to be someone who says, I want 3! xD then where does that stop. I do think limiting it to a secondary camera would cover most basis if not all.

It makes sense to have a switch, toggle between and all cameras.

There's some other camera work I'm planning to sort out, ie... the aweful fullscreen implementation that's available now. I will add this to the list to consider when I get around to that. It's probably going to be after V1.2 which I'm working on at the moment. I'd add it but it's already a rather large overhaul and I wanna get it out the door ahaha.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by NotExpectedYet** [comment]
> Can you even do three cameras? I guess you could do 2 USB and a RasPi. I never thought about it. But yeah, a limit of two would be good since most people probably self limit to 2 cause the resources needed for more pretty much limits you to a high end Pi4. I have two 3B+ and they handle it ok.  Thanks for considering the feature, and no rush. 😁Knowing it's a future possibility is good.  When you get around to it, I'll happily beta for you.  

Mine are all RTSP external to the Pi so in theory I could quite easily. Getting mjpeg streamer to run multiple usb camera's is a challenge on it's own so I doubt any would with Pi per printer. 

No trouble at all! I've opened a project for tracking it. 
https://github.com/OctoFarm/OctoFarm/projects/34

**Reply by dirkvranckaert** [reply]
Is there any progress to be expected on this?
